src/cascadia/TerminalApp/App.cpp

```diff
@@ -25,6 +25,67 @@ namespace winrt
     using IInspectable = Windows::Foundation::IInspectable;
 }
 
+// clang-format off
+static const std::unordered_map<::TerminalApp::SettingsLoadWarnings, std::wstring_view> settingsLoadWarningsLabels {
+    { SettingsLoadWarnings::MissingDefaultProfile , L"MissingDefaultProfileText"},
+    { SettingsLoadWarnings::DuplicateProfile      , L"DuplicateProfileText"},
+};
+static const std::unordered_map<::TerminalApp::SettingsLoadErrors, std::wstring_view> settingsLoadErrorsLabels {
+    { SettingsLoadErrors::NoProfiles              , L"NoProfilesText"},
+};
+// clang-format on
+
+// Function Description:
+// - General-purpose helper for looking up a localized string for a
+//   warning/error. First will look for the given key in the provided map of
+//   keys->strings, where the values in the map are ResourceKeys. If it finds
+//   one, it will lookup the localized string from that ResourceKey.
+// - If it does not find a key, it'll return an empty string
+// Arguments:
+// - key: the value to use to look for a resource key in the given map
+// - map: A map of keys->Resource keys.
+// - loader: the ScopedResourceLoader to use to look up the localized string.
+// Return Value:
+// - the localized string for the given type, if it exists.
+template<typename T>
+static winrt::hstring _GetMessageText(T key, std::unordered_map<T, std::wstring_view> map, ScopedResourceLoader loader)

```

Comment:
consider making the maps #972 compliant.

Reply:
how though? I tried just adding `std::less<>` a template arg to the map declarations, and that gave me all sorts of trouble  about `operator __surrogate_func` and `less<void>` expectind 2 args but found one or something. Then I tried doing the `AppKeyBindings` thing with `KeyChordHash, KeyChordEquality`, but I couldn't get the two functions below to be happy calling `_GetMessageText` because the map wasn't the same type or something. 

I think I'm gonna loop back on this with the array solution above

Reply:
Ah i believe they only work for the _non_ `unordered_` versions of containers in C++14/17